tests/generic_config_updater/gu_utils.py

```diff
@@ -366,3 +372,91 @@ def is_valid_platform_and_version(duthost, table, scenario, operation, field_val
         return False
     except IndexError:
         return False
+
+
+def format_and_apply_template(duthost, template_name, extra_vars):
+    dest_path = os.path.join(TMP_DIR, template_name)
+    duthost.host.options['variable_manager'].extra_vars.update(extra_vars)
+    duthost.file(path=dest_path, state='absent')
+    duthost.template(src=os.path.join(TEMPLATES_DIR, template_name), dest=dest_path)
+
+    # duthost.template uses single quotes, which breaks apply-patch. this replaces them with double quotes
+    duthost.shell("sed -i \"s/'/\\\"/g\" " + dest_path)
+
+    output = duthost.shell("config apply-patch {}".format(dest_path))
+
+    duthost.file(path=dest_path, state='absent')
```

Comment:
Better to surround with `try catch` to ensure file is deleted.

Reply:
Surrounded duthost shell calls with with try/catch block and moved file deletion to finally block